AdyenComponents/Apple Pay/ApplePayComponentExtensions.swift

```diff
@@ -137,6 +105,31 @@ extension ApplePayComponent {
 
         return networks
     }
+
+    internal static func supportedNetworks(for paymentMethod: ApplePayPaymentMethod) -> [PKPaymentNetwork] {
+        var networks = ApplePayComponent.defaultNetworks
+        if let brands = paymentMethod.brands {
+            let brandsSet = Set(brands)
+            networks = networks.filter { brandsSet.contains($0.adyenName) }
+        }
+        return networks
+    }
```

Comment:
I think like the `supportedNetworks `, `payment `, `paymentRequest`, and `paymentMethod ` in configuration.

why is this change exactly? what's wrong with having them in configuration?

Reply:
The main goal is to do 

```swift
    return try PreApplePayComponent(paymentMethod: paymentMethod,
                                    apiContext: apiContext,
                                    payment: payment,
                                    configuration: applePay)
```

Instead of 

```swift

let configuration = ApplePayComponent.Configuration(
    payment: payment,
    paymentMethod: paymentMethod,
    summaryItems: applePay.summaryItems,
    merchantIdentifier: applePay.merchantIdentifier,
    requiredBillingContactFields: applePay.requiredBillingContactFields,
    requiredShippingContactFields: applePay.requiredShippingContactFields
)

return try PreApplePayComponent(apiContext: apiContext, configuration: configuration)

```

### `supportedNetworks` 
is basically a property of `ApplePayPaymentMethod` (moved).

### `payment` 
could be part of Configuration, sure. 
The only issue - payment is required parameter and it leads to assign it twice

```swift
let configuration = DropInComponent.Configuration(apiContext: apiContext)
        configuration.applePay = .init(payment: payment,
                                       summaryItems: ConfigurationConstants.applePaySummaryItems,
                                       merchantIdentifier: ConfigurationConstants.applePayMerchantIdentifier)
        configuration.payment = payment
```

### `paymentMethod`
could be part of Configuration, sure. 
I was trying to align ApplePay with other payment methods. common pattern is 

```swift
let component = SOMEComponent(paymentMethod: paymentMethod,
                              apiContext: apiContext,
                              style: style)
```

### `paymentRequest `
moved outside of `Configuration` simply to keep `Configuration` a plain object.